src/verify.py

```diff
@@ -0,0 +1,533 @@
+#!/usr/bin/env python3
+
+"""Generate and run test content"""
+
+import common
+import video_generate
+import audio_generate
+import audio_common
+import video_common
+import vft
+import pandas as pd
+import sys
+import argparse
+import tempfile
+import cv2
+import numpy as np
+import graycode
+import math
+import os
+import scipy
+import inspect
+import glob
+
+
+BEEP_PERIOD_SEC = 3.0
+WIDTH = 720
+HEIGHT = 536
+FPS = 30
+VFT_LAYOUT = vft.VFT_LAYOUT[vft.DEFAULT_VFT_ID]
+BEEP_PERIOD_FRAMES = BEEP_PERIOD_SEC * FPS
+
+DURATION_SEC = 12
+NUM_FRAMES = int(DURATION_SEC * FPS)
+SAMPLE_RATE = 16000
+DEBUG = 0
+PREC = 0.01  # sec
+KEEP_FILES = False
+
+
+def audio_generate(duration_sec, output_filename, **settings):
+    audio_delay = settings.get("audio_delay", 0)
+    video_delay = settings.get("video_delay", 0)
+    samplerate = settings.get("samplerate", SAMPLE_RATE)
+
+    # normal default signal
+    aud = audio_common.generate_chirp(duration_sec, **settings)
+
+    audio_signal = None
+    auddio_filtered = None
+    if audio_delay > 0:
+        delay = [0.0] * int(audio_delay * samplerate)
+        audio_filtered = np.concatenate((delay, aud))[: len(aud)]
+    elif audio_delay < 0:
+        audio_filtered = aud[-int(audio_delay * samplerate) :]
+    else:
+        audio_filtered = aud
+
+    # video delay cannot be negative...
+    if video_delay > 0:
+        delay = [0.0] * int(video_delay * samplerate)
+        tmp = aud.copy()[int(video_delay * samplerate) :]
+        audio_filtered = audio_filtered // 4
+        audio_signal = np.concatenate((tmp, delay))[: len(aud)]
+
+        aud = (audio_filtered + audio_signal) // 2
+    else:
+        aud = audio_filtered
+
+    scipy.io.wavfile.write(
+        output_filename, audio_common.DEFAULT_SAMPLERATE, aud.astype(np.int16)
+    )
+
+
+OLD_FRAME = None
+BLACK_FRAME = None
+
+
+def write_frame(rawstream, frame, frame_num, freeze_frames, black_frames):
+    global OLD_FRAME
+    global BLACK_FRAME
```

Comment:
in general, global variables are messy (code is harder to read, and may create unexpected code flows). Can they be replaced?

Reply:
Removed.